controllers/dns_helper.go

```diff
@@ -206,7 +205,7 @@ func (dh *dnsHelper) getSimpleEndpoints(mcgTarget *multicluster.GatewayTarget, h
 // ab2.lb-a1b2.shop.example.com A 192.22.2.3
 // ab3.lb-a1b2.shop.example.com A 192.22.2.4
 
-func (dh *dnsHelper) getLoadBalancedEndpoints(mcgTarget *multicluster.GatewayTarget, hostname string, currentEndpoints map[string]*externaldns.Endpoint) []*externaldns.Endpoint {
+func (dh *dnsHelper) getLoadBalancedEndpoints(mcgTarget *multicluster.GatewayTarget, hostname string, currentEndpoints map[string]*externaldns.Endpoint, clusterID string) []*externaldns.Endpoint {
```

Comment:
The idea here was that `mcgTarget` had everything set on it required to generate endpoints. So in here it would either use `GetName` or `GetShortCode` on the mcgTarget (the outer multicluster gateway), or the cgwTarget(the single cluster gateway).

You'd probably be better to add `clusterID` as the name of the cluster on the [ClusterGateway](https://github.com/Kuadrant/kuadrant-operator/blob/main/pkg/multicluster/gateway_wrapper.go#L153) so that its used when you call `cgwTarget.GetShortCode()` or `cgwTarget.GetName()`

Also see comment from Craig, the `GetName` method of the cluster target probably needs to include the gw name and namespace similar to [this](https://github.com/Kuadrant/kuadrant-operator/blob/main/pkg/multicluster/target.go#L37)  so, `<clusterID>`-`<GW Name>`-`<GW Namespace>`.